shared/middleware/helpers/property_bag.rb

```diff
@@ -30,6 +31,8 @@ def get(name)
   end
 
   def set(name, value, ip_address)
+    raise NotFound unless @can_modify
```

Comment:
add:

```
def forbidden!()
  halt(403, "Forbidden\n")
end
```

to `shared/middleware/helpers/core.rb` and rename `raise NotFound` with forbidden!.